be/src/util/frame_of_reference_coding.h

```diff
@@ -0,0 +1,151 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef DORIS_FRAME_OF_REFERENCE_CODING_H
+#define DORIS_FRAME_OF_REFERENCE_CODING_H
+
+
+#include <cstdlib>
+#include <iostream>
+
+#include "util/bit_stream_utils.h"
+#include "util/bit_stream_utils.inline.h"
+#include "util/faststring.h"
+
+namespace doris {
+
+struct ForCoding {
+    const static uint8_t FRAME_VALUE_NUM = 128;
+};
+
+// The implementation for frame-of-reference coding
+// The detail of frame-of-reference coding, please refer to
+// https://lemire.me/blog/2012/02/08/effective-compression-using-frame-of-reference-and-delta-coding/
+// and https://www.elastic.co/cn/blog/frame-of-reference-and-roaring-bitmaps
+
+// The encoded data format is as follows:
+
+// 1. Body:
+// BitPackingFrame * FrameCount
+// 2. Footer:
+// BitWidth(1) * FrameCount
+// OrderFlag(1) * FrameCount
+// LastFrameNumValues(1) (optional)
+// FrameCount(4)
+// NumValues(4)
```

Comment:
I think it's better to store FrameSize(currently 128, could be changed in the future) and NumValues, then FrameCount and LastFrameNumValues can be calculated.

Reply:
OK